File: scala/unwrittenfun/minecraft/unwrittenblocks/blocks/BlockDarkInfuser.scala
package unwrittenfun.minecraft.unwrittenblocks.blocks

import net.minecraft.block.BlockContainer
import net.minecraft.block.material.Material
import net.minecraft.world.World
import net.minecraft.tileentity.TileEntity
import net.minecraft.client.renderer.texture.IconRegister
import unwrittenfun.minecraft.unwrittenblocks._
import unwrittenfun.minecraft.unwrittenblocks.blocks.tileentities.TileEntityDarkInfuser
import cpw.mods.fml.common.network.FMLNetworkHandler
import net.minecraft.entity.player.EntityPlayer
import unwrittenfun.minecraft.unwrittenblocks.client.gui.creativetabs

/**
 * Mod: UnwrittenBlocks
 * Author: UnwrittenFun
 * License: Minecraft Mod Public License (Version 1.0.1)
 */
class BlockDarkInfuser(id: Int, key: String) extends BlockContainer(id, Material.iron) {
  setUnlocalizedName(key)
  setHardness(2f)
  setCreativeTab(creativetabs.ubTab)

  def createNewTileEntity(world: World): TileEntity = new TileEntityDarkInfuser

  override def registerIcons(register: IconRegister) {
    blockIcon = register.registerIcon(TEXTURE_LOCATION + ":" + key)
  }

  override def onBlockActivated(world: World, x: Int, y: Int, z: Int, player: EntityPlayer, side: Int, hitX: Float, hitY: Float, hitZ: Float): Boolean = {
    world.getBlockTileEntity(x, y, z) match {
      case infuser: TileEntityDarkInfuser => if (!world.isRemote) FMLNetworkHandler.openGui(player, UnwrittenBlocks, 1, world, x, y, z); true
      case _ => false
    }
  }

  override def breakBlock(world: World, x: Int, y: Int, z: Int, id: Int, meta: Int) {
    UBBlocks.dropItemsFromInventory(world, x, y, z)
    super.breakBlock(world, x, y, z, id, meta)
  }
}

File: src/main/java/unwrittenfun/minecraft/unwrittenblocks/common/blocks/BlockDarkInfuser.java
package unwrittenfun.minecraft.unwrittenblocks.common.blocks;

import cpw.mods.fml.common.network.internal.FMLNetworkHandler;
import net.minecraft.block.BlockContainer;
import net.minecraft.block.material.Material;
import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.tileentity.TileEntity;
import net.minecraft.world.World;
import unwrittenfun.minecraft.unwrittenblocks.common.ModInfo;
import unwrittenfun.minecraft.unwrittenblocks.common.UnwrittenBlocks;
import unwrittenfun.minecraft.unwrittenblocks.common.tileEntities.TEDarkInfuser;

/**
 * Project: UnwrittenBlocks Author: UnwrittenFun Created: 04/11/2014.
 */
public class BlockDarkInfuser extends BlockContainer {
  protected BlockDarkInfuser(String key) {
    super(Material.rock);
    setCreativeTab(UnwrittenBlocks.creativeTabUB);
    setBlockName(key);
    setBlockTextureName(ModInfo.RESOURCE_LOCATION + ":" + key);
    setHardness(2F);
    setBlockBounds(9F / 32F, 0F, 9F / 32F, 23F / 32F, 30F / 32F, 23F / 32F);
  }

  @Override
  public TileEntity createNewTileEntity(World p_149915_1_, int p_149915_2_) {
    return new TEDarkInfuser();
  }

  @Override
  public boolean renderAsNormalBlock() {
    return false;
  }

  @Override
  public int getRenderType() {
    return -1;
  }

  @Override
  public boolean isOpaqueCube() {
    return false;
  }

  @Override
  public boolean onBlockActivated(World world, int x, int y, int z, EntityPlayer player, int side, float hitX,
                                  float hitY, float hitZ) {
    TileEntity tileEntity = world.getTileEntity(x, y, z);
    if (tileEntity instanceof TEDarkInfuser) {
      FMLNetworkHandler.openGui(player, UnwrittenBlocks.instance, 1, world, x, y, z);
      return true;
    }
    return false;
  }
}
